server/src/main/java/org/elasticsearch/index/translog/Checkpoint.java

```diff
@@ -172,9 +172,8 @@ public static void write(ChannelFactory factory, Path checkpointFile, Checkpoint
         // now go and write to the channel, in one go.
         try (FileChannel channel = factory.open(checkpointFile, options)) {
             Channels.writeToChannel(bytes, channel);
-            // no need to force metadata, file size stays the same and we did the full fsync
-            // when we first created the file, so the directory entry doesn't change as well
-            channel.force(false);
+            // fsync with metadata as we use this method when creating the file
+            channel.force(true);
```

Comment:
This is also used by `TranslogWriter.writeCheckpoint()`, where there is no need to fsync the metadata (and we should not for performance reasons).

Reply:
I think that uses the other signature with the file channel passed? This method is only used by the 3 production code spots I adjusted in this PR as far as I can tell:

![image](https://user-images.githubusercontent.com/6490959/99664207-1f002280-2a68-11eb-8cc6-c6486de02d59.png)